### DISCUSSION TITLE
Set default merge branch for PR to base branch

### DISCUSSION THREAD

**Discussion by benavrhm** [community/community#23578] [New to GitHub]
<p>I have a public mirror repo that uses the “master” branch to track an upstream master branch in an Open Source project that I do not manage. The master branch in my repo is therefore protected and is not used for merging local development branches.</p>
<p>My repo contains several local branches that represent various product releases, so we call them “product branches”. Development “topic branches” are made by branching from one or another of these product branches. The merge branch for each topic branch pull request (PR) should always be the product branch from which the topic branch was created, never the “master” branch.</p>
<p>Is there a way to configure the default merge branch for PR’s depending on the branch history of the PR branch?</p>
<p>For example, if my repo has the following branches:</p>
<ol>
<li>master</li>
<li>tomato</li>
<li>potato</li>
</ol>
<p>The development topic branches will be branched only from “tomato” or “potato”, never from “master”, and a topic branch will always be merged back to the branch from which it came. So if a developer branched a topic branch “potato-Jira-553” from “potato”, then the default merge branch for the PR should be set to “potato” and not to “master”.</p>

**Comment by jsoref** [comment, marked as answer]
<p>Not really.</p>
<p>With some effort you could have two repositories <code>tomato</code> and <code>potato</code>.</p>
<p>They could each be set to have a single upstream branch <code>master</code>.</p>
<p>And each could have a default branch of <code>tomato</code> and <code>potato</code> respectively.</p>
<p>The default branch for a PR is the default branch of the repository. GitHub doesn’t really pay any attention to minutiae regarding Git graph structure.</p>
<p>Only one of your repositories could be a GitHub connected fork, but in this case, you could probably make <em>neither</em> of them GitHub connected forks – possibly using a third repository to detect updates in the upstream fork and automatically push those commits to the <code>master</code> branch in both.</p>

**Comment by benavrhm** [comment]
<p>TNX.</p>
<p>Looks like splitting into two repos and managing the upstream updates for both from a third repo is the only solution given that there is apparently no fine-grain control over the GitHub PR workflow.</p>
